### DISCUSSION TITLE
How do I package whisper with PyInstaller

### DISCUSSION THREAD

**Discussion by hewenzhuo1** [openai/whisper#1479] [Q&A]
I tried to package the program containing whisper using PyInstaller but he reported an error.

![1688022234879](https://github.com/openai/whisper/assets/45852301/39f185f1-4187-4af1-a1d1-89b50f3cc61e)

Is there an expert who can help me with what I should do！

**Comment by ururk** [comment]
In my .spec file, I had to manually include the assets folder:

```python
    datas=[
        ('env/lib/python3.10/site-packages/whisper/assets', './whisper/assets')
    ],
```

There may be a slicker way of doing this with pyinstaller.

Also - you will probbaly have to declare your main function like this:

```python
if __name__ == '__main__':
	# Pyinstaller fix
	multiprocessing.freeze_support()

	main()
```

Without `freeze_support`, it just loops endlessly.

**Reply by Kuprich** [reply]
Oh, Thanks a lot!

in my .spec file:
```
datas = [('C:\\Users\\admin\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\whisper', './whisper')]
``` 
and I didn't make any more changes..

next step:
```
pyinstaller main.spec
```

**Comment by RavinMaddHatter** [comment]
I think this worked back in 2023. but now i get 

`'C:\\Users\\***\\AppData\\Local\\Temp\\****\\whisper\\assets\\multilingual.tiktoken'`

Any ideas on how to fix that?
